### PR TITLE
Single segment motorway check

### PR DESCRIPTION
**Pull request by mm-ciub**
### Description:

Added fix suggestion for the SingleSegmentMotorwayCheck (https://github.com/osmlab/atlas-checks/issues/407).
The suggestion will only be applied to 100% sure cases where all other connected roads have the same highway tag value.

### Potential Impact:

No impact.

### Unit Test Approach:

Existing unites have been updated to also look for the expected fix suggestion where applicable.

### Test Results:

Manually checked cases to verify that the fix suggestion works as expected.
Also tried it on ROU, NZL and SGP with no flags. 

| ISO | Total Flags | Sampled  | Sampling % | TP | FP | False Positive Rate |
|-----|-------------|----------|------------|----|----| ------------------- |
|  AUS   |    4         |    4      |     100%       |   4 |   0 | 0%   |
|  VNM   |    4         |    4      |     100%       |   4 |   0 | 0%   |
|  PHL   |    4         |    4      |     100%       |   4 |   0 | 0%   |

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java [hunk 1]
```diff
@@ -73,8 +78,14 @@ public boolean validCheckForObject(final AtlasObject object)
                 .noneMatch(this::isMotorwayNotRoundabout))
         {
             this.markAsFlagged(edge.getOsmIdentifier());
-            return Optional.of(this.createFlag(new OsmWayWalker(edge).collectEdges(),
-                    this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+            final CheckFlag flag = this.createFlag(new OsmWayWalker(edge).collectEdges(),
+                    this.getLocalizedInstruction(0, edge.getOsmIdentifier()));
+            final Optional<String> fixSuggestion = this.getFixSuggestion(edge);
+            return fixSuggestion.map(s -> flag.addFixSuggestion(FeatureChange.add(
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java:84]: Please use a more descriptive variable name than `s`.

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java [hunk 2]
```diff
@@ -24,23 +25,48 @@ public void invalidMotorwaySegmentOneConnectionRoundaboutTest()
     {
         this.verifier.actual(this.setup.invalidMotorwaySegmentOneConnectionRoundaboutAtlas(),
                 new SingleSegmentMotorwayCheck(ConfigurationResolver.emptyConfiguration()));
-        this.verifier.globallyVerify(flags -> Assert.assertEquals(1, flags.size()));
+        this.verifier.globallyVerify(flags ->
+        {
+            Assert.assertEquals(1, flags.size());
+            Assert.assertEquals(1, flags.get(0).getFixSuggestions().size());
+            flags.get(0).getFixSuggestions()
+                    .forEach(s -> Assert.assertTrue(
```
Comment (Bentleysb) [inline src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java:33]: Again, please use a more descriptive variable name than a single character.

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java [hunk 3]
```diff
@@ -24,23 +25,48 @@ public void invalidMotorwaySegmentOneConnectionRoundaboutTest()
...
     {
         this.verifier.actual(this.setup.invalidMotorwaySegmentOneConnectionAtlas(),
                 new SingleSegmentMotorwayCheck(ConfigurationResolver.emptyConfiguration()));
-        this.verifier.globallyVerify(flags -> Assert.assertEquals(1, flags.size()));
+        this.verifier.globallyVerify(flags ->
+        {
+            Assert.assertEquals(1, flags.size());
+            Assert.assertEquals(1, flags.get(0).getFixSuggestions().size());
+            flags.get(0).getFixSuggestions()
+                    .forEach(s -> Assert.assertTrue(
```
Comment (Bentleysb) [inline src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java:49]: Same as above.

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java [hunk 4]
```diff
@@ -24,23 +25,48 @@ public void invalidMotorwaySegmentOneConnectionRoundaboutTest()
...
     {
         this.verifier.actual(this.setup.invalidPrimaryMotorwayPrimarySegmentAtlas(),
                 new SingleSegmentMotorwayCheck(ConfigurationResolver.emptyConfiguration()));
-        this.verifier.globallyVerify(flags -> Assert.assertEquals(1, flags.size()));
+        this.verifier.globallyVerify(flags ->
+        {
+            Assert.assertEquals(1, flags.size());
+            Assert.assertEquals(1, flags.get(0).getFixSuggestions().size());
+            flags.get(0).getFixSuggestions()
+                    .forEach(s -> Assert.assertTrue(
```
Comment (Bentleysb) [inline src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java:66]: Same as above.

**Review by Bentleysb — COMMENTED** [review]
This looks great! There are just a few single character variable names that need to change.

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java [hunk 5]
```diff
@@ -85,6 +96,14 @@ public boolean validCheckForObject(final AtlasObject object)
         return FALLBACK_INSTRUCTIONS;
     }
 
+    private Optional<String> getFixSuggestion(final Edge edge)
+    {
+        final Set<String> options = edge.connectedEdges().stream().filter(Edge::isMainEdge)
+                .map(HighwayTag::highwayTag).flatMap(Optional::stream).map(Enum::name)
```
Comment (seancoulter) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java:102]: nice use of Optional::stream

**Review by seancoulter — COMMENTED** [review]
LGTM, once @Bentleysb 's comments are addressed

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java (hunk 1 above)
Comment (mm-ciub) [reply src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheck.java]: Renamed to suggestion.

File: src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java (hunk 2 above)
Comment (mm-ciub) [reply src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/SingleSegmentMotorwayCheckTest.java]: Renamed to fixSuggestion.

**Review by Bentleysb — APPROVED** [review]

**Review by seancoulter — APPROVED** [review]

**Pull request merged**